File: Python/chapter 3/1253.py
num = int(input())
num_list = list(map(int, input().split()))
num_list.sort()
count = 0

for i in range(num):
    find = num_list[i]
    start_idx = 0
    end_idx = num -1

    while(start_idx < end_idx):
        if(num_list[start_idx] + num_list[end_idx] == find):
            if(start_idx != i and end_idx != i): # 중복 값이 있을 수 있으므로 idx로 검사 필요 !!!!! 
                count += 1
                break   
            elif(start_idx == i):
                start_idx += 1
                
            elif(end_idx == i):
                end_idx -= 1
        elif(num_list[start_idx] + num_list[end_idx] > find):
            end_idx -= 1
        elif(num_list[start_idx] + num_list[end_idx] < find):
            start_idx += 1

print(count)

File: CPP/chapter3/1253.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main(){
    int num;
    cin >> num;

    vector<int> numList(num,0);

    for(int i=0;i<num;i++){
        cin >> numList[i];
    }

    sort(numList.begin(), numList.end());

    int result = 0;

    for(int k = 0 ; k < num ; k++){
        long find = numList[k];
        int i = 0;
        int j = num - 1;

        while( i < j){
            if(numList[i] + numList[j] == find){
                if(i != k && j != k){
                    result++;
                    break;
                }else if(i == k){
                    i++;
                }else if (j == k){
                    j--;
                }
            }else if(numList[i] + numList[j] < find){
                i ++;
            }else{
                j--;
            }
            
        }

    }

    cout << result;

}